tempodb/search/rescan_blocks.go

```diff
@@ -27,41 +26,48 @@ func RescanBlocks(walPath string) ([]*StreamingSearchBlock, error) {
 
 	blocks := make([]*StreamingSearchBlock, 0, len(files))
 	for _, f := range files {
-		if f.IsDir() {
+		info, err := f.Info()
+
+		if err != nil {
+			level.Warn(log.Logger).Log("msg", "failed to file info", "err", err)
```

Comment:
we're logging this as a warning but treating it as a fatal error by bailing out. if this error hits let's just return it instead of logging it and fail the whole rescan process.

```
return nil, fmt.Errorf("failed to file info %s, %w", f.NameORSomething, err)
```